proto/public.proto

```diff
@@ -339,6 +340,15 @@ message BasicStats {
   uint64 actual_failure_count = 8;
 }
 
+message TcpStats {
+  // number of currently open connections
+  uint64 open_connections = 1;
```

Comment:
thoughts around including these tcp stats in `BasicStats`?

taking that a step further, what about always including tcp stats in stats queries? pros and cons that come to mind:
- always include
  - simpler code
  - aligns with flatter telemetry model in proxies/prometheus/grafana
- require `tcp_stats=true`
  - potentially less pressure on prometheus?
  - allows for more/different types of stats going forward?
  - maybe tcp stats aren't a thing for some stats queries?

Reply:
Yeah totally, I did think about that! So currently, I'm adding 3 new prometheus queries (and when we add duration it will be 4) which is doubling the number of queries sent per stat request. I wasn't sure whether that was worth it, unless we were committed to always displaying TCP stats everywhere (since our current views don't use them, it seemed like a lot to double the queries for every case, when we don't surface the stats in many places). However, I could also see us incorporating TCP stats into normal stat views, where I agree it could make sense to always have them.

Reply:
Test results show no significant increase in resource usage when always querying for TCP metrics. The only variation between baseline and the "always TCP" case is an increase in queries to Prometheus, which is expected.

# Results

## Prometheus mem/cpu/latency

### Baseline

<img width="1053" alt="baseline prometheus" src="https://user-images.githubusercontent.com/236915/52735001-685e3c80-2f7b-11e9-90d5-ff2c80f3e86e.png">

### Always tcp

<img width="1060" alt="always-tcp prometheus" src="https://user-images.githubusercontent.com/236915/52735000-685e3c80-2f7b-11e9-9a23-5b18fb4231c7.png">

## Prometheus linkerd-proxy stats

### Baseline

<img width="1047" alt="baseline prom-proxy-metrics" src="https://user-images.githubusercontent.com/236915/52734999-67c5a600-2f7b-11e9-9a37-3b8d3ffbff20.png">

### Always tcp

<img width="1060" alt="always-tcp prom-proxy-metrics" src="https://user-images.githubusercontent.com/236915/52734998-67c5a600-2f7b-11e9-835d-3fb401ffdb90.png">

# Test Setup

- Linkerd built off of this branch
- deployed to 2 namespaces, `l5d-baseline`, and `l5d-always-tcp`
- 2 emojivoto namespaces created, one for each linkerd. each had 10 replicas per-deployment, so each Linkerd was managing 40 injected pods
- 15-node AKS cluster, 4 vcpus, 14 GB memory per node
- 15 hour run, followed by evaluation of metrics in Prometheus

## Scripts

These scripts were run from a Kubernetes node in a tight bash `while... do ` loop.

### Baseline

```bash
#!/bin/bash

curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=all&all_namespaces=true&window=1m"
curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=namespace&all_namespaces=true&window=1m"
curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=pod&namespace=emojivoto-baseline&tcp_stats=true&window=1m"
curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=deployment&namespace=emojivoto-baseline&resource_name=emoji&window=1m"
curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=all&all_namespaces=true&to_name=emoji&to_type=deployment&to_namespace=emojivoto-baseline&window=1m"
curl -s -o /dev/null "http://10.0.137.105:8084/api/tps-reports?resource_type=all&all_namespaces=true&from_name=emoji&from_type=deployment&from_namespace=emojivoto-baseline&window=1m"
```

### Always tcp

```
#!/bin/bash

curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=all&all_namespaces=true&window=1m&tcp_stats=true"
curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=namespace&all_namespaces=true&window=1m&tcp_stats=true"
curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=pod&namespace=emojivoto-always-tcp&tcp_stats=true&window=1m"
curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=deployment&namespace=emojivoto-always-tcp&resource_name=emoji&window=1m&tcp_stats=true"
curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=all&all_namespaces=true&to_name=emoji&to_type=deployment&to_namespace=emojivoto-always-tcp&window=1m&tcp_stats=true"
curl -s -o /dev/null "http://10.0.122.86:8084/api/tps-reports?resource_type=all&all_namespaces=true&from_name=emoji&from_type=deployment&from_namespace=emojivoto-always-tcp&window=1m&tcp_stats=true"
```

Reply:
based on this info, I'm going to move TcpStats into BasicStats, and always query for TCP stats on stat queries